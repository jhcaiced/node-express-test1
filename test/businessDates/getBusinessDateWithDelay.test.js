const config = require('../../src/config')
const request = require('supertest')(`${config.url}/api/v1/businessDates/`)

describe('Settlement Transfer Date Estimate', () => {
  it('Should calculate a Settlement Transfer Date Estimate', async () => {
    let response = await request.post('getBusinessDateWithDelay').send({
      initialDate: '2018-11-10T10:10:10Z',
      delay: 3
    }).expect(200)
    expect(response.header['content-type']).toBe('application/json; charset=utf-8')
    expect(response.text).toMatchSnapshot()
    response = await request.post('getBusinessDateWithDelay').send({
      initialDate: '2018-11-15T10:10:10Z',
      delay: 3
    })
    expect(response.text).toMatchSnapshot()
    response = await request.post('getBusinessDateWithDelay').send({
      initialDate: '2018-12-25T10:10:10Z',
      delay: 20
    })
    expect(response.text).toMatchSnapshot()
  })
})
