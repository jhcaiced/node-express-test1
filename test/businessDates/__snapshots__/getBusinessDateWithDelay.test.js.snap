// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settlement Transfer Date Estimate Should calculate a Settlement Transfer Date Estimate 1`] = `"{\\"ok\\":true,\\"initialQuery\\":{\\"initialDate\\":\\"2018-11-10T10:10:10Z\\",\\"delay\\":3},\\"results\\":{\\"businessDate\\":\\"2018-11-10T10:10:10Z\\",\\"totalDays\\":0,\\"holidayDays\\":0,\\"weekendDays\\":0}}"`;

exports[`Settlement Transfer Date Estimate Should calculate a Settlement Transfer Date Estimate 2`] = `"{\\"ok\\":true,\\"initialQuery\\":{\\"initialDate\\":\\"2018-11-15T10:10:10Z\\",\\"delay\\":3},\\"results\\":{\\"businessDate\\":\\"2018-11-15T10:10:10Z\\",\\"totalDays\\":0,\\"holidayDays\\":0,\\"weekendDays\\":0}}"`;

exports[`Settlement Transfer Date Estimate Should calculate a Settlement Transfer Date Estimate 3`] = `"{\\"ok\\":true,\\"initialQuery\\":{\\"initialDate\\":\\"2018-12-25T10:10:10Z\\",\\"delay\\":20},\\"results\\":{\\"businessDate\\":\\"2018-12-25T10:10:10Z\\",\\"totalDays\\":0,\\"holidayDays\\":0,\\"weekendDays\\":0}}"`;
